File: client/src/app/models/Product.ts
import { ProductPhoto } from "./ProductPhoto";
import { ProductReview } from "./productReview";
import { Accessories } from "./Products Specifcation/Accessories";
import { ComputerAndLaptop } from "./Products Specifcation/computer-and-laptop";
import { Mobile } from "./Products Specifcation/Mobile";
import { MonitorProduct } from "./Products Specifcation/MonitorProduct";
import { Pet } from "./Products Specifcation/Pet";
import { Vehicle } from "./Products Specifcation/Vehicle";

export interface Product {
    id: number;
    productName: string;
    category: string;
    subCategory: string;
    link: string;
    price: number;
    onWishlist: boolean;
    onCart: boolean;
    quantity: number;
    sellerId: string;
    sellerName: string;
    productDescription: string;
    productPhotos: ProductPhoto[];
    state: string;
    created: string;
    computerAndLaptop: ComputerAndLaptop;
    vehicle: Vehicle;
    pet: Pet;
    mobile: Mobile;
    accessories: Accessories;
    monitorProduct: MonitorProduct;
    specification: string;
    productReview: ProductReview;
    reported: boolean;
    discountRate: number;
    priceAfterDiscount: number;
}

File: API/Shopx.API/Entities/Product.cs
using Shopx.API.Data;
using Shopx.API.Entities.Product_Specification;
using System.ComponentModel.DataAnnotations.Schema;

namespace Shopx.API.Entities
{
    [Table("Products")]
    public class Product
    {
        public int Id { get; set; }
        public string ProductName { get; set; }///product title
        public string Category { get; set; }
        public string SubCategory { get; set; }
        public double Price { get; set; }
        public int DiscountRate { get; set; }
        public int Quantity { get; set; }
        public string SellerId { get; set; }
        public string SellerName { get; set; }
        public string ProductDescription { get; set; }
        public string State { get; set; } = States.active;
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public AppUser Seller { get; set; }
        public List<Sales> BoughtProduct { get; set; }
        public List<ShoppingCart> Carts { get; set; }
        public List<ProductPhoto> ProductPhotos { get; set; } = new();
        public List<ProductReview> ProductReviews { get; set; }
        public List<BrowseHistory> BrowseHistories { get; set; }
        public List<ReportProduct> ReportProducts { get; set; }
        public int ProductViews { get; set; }

        ////product specification
        public int? LaptopId { get; set; }
        public LaptopAndComputer Laptop { get; set; }///laptop and computer

        public int? VehicleId { get; set; }
        public Vehicle Vehicle { get; set; }

        public int? PetId { get; set; }
        public Pet Pet { get; set; }

        public int? MobileId { get; set; }
        public Mobile Mobile { get; set; }

        public int? AccessoriesId { get; set; }
        public Accessories Accessories { get; set; }

        public int? MonitorProductId { get; set; }
        public MonitorProduct MonitorProduct { get; set; }
    }
}
